Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm

```diff
@@ -965,10 +965,12 @@ static void decodeInvocationArguments(WKRemoteObjectDecoder *decoder, NSInvocati
         case '@': {
             auto& allowedClasses = allowedArgumentClasses[i - firstArgument];
 
-            id value = decodeObjectFromObjectStream(decoder, allowedClasses);
-            [invocation setArgument:&value atIndex:i];
+            RetainPtr value = decodeObjectFromObjectStream(decoder, allowedClasses);
+            SUPPRESS_UNRETAINED_LOCAL id valueId = value.get();
+            [invocation setArgument:&valueId atIndex:i];
 
-            // FIXME: Make sure the invocation doesn't outlive the value.
+            // We need to make sure the invocation keep the value argument alive.
+            [invocation retainArguments];
```

Comment:
@darinadler @geoffreygaren I fixed the crashes by asking `invocation` to retain its arguments here. The previous code was super fragile. The `invocation` would not keep its parameter alive and essentially would hope the `value` stayed alive long enough simply because it was autoreleased.

Because I moved from autorelease to RetainPtr, the value would get destroyed right away and the invocation would outlive its argument.

Are you OK with this fix?